src/rust/engine/process_execution/src/lib.rs

```diff
@@ -225,13 +229,19 @@ pub struct Process {
   #[derivative(PartialEq = "ignore", Hash = "ignore")]
   pub description: String,
 
-  // This will be materialized for local Process only.
-  // Eventually we want to remove this.
-  // Context: https://github.com/pantsbuild/pants/issues/8314
-  // Think twice before using it.
-  #[derivative(PartialEq = "ignore", Hash = "ignore")]
-  pub unsafe_local_only_files_because_we_favor_speed_over_correctness_for_this_rule:
-    hashing::Digest,
+  ///
+  /// Declares that this process uses the given named caches (which might have associated config
+  /// in the future). Cache names must contain only lowercase ascii characters or underscores.
+  ///
+  /// Caches are exposed to processes within their workspaces as (symlinks to) directories at the
```

Comment:
Weird, github apparently didn't post my line comment :S

I don't think that we should specify that this will be a symlink. It may happen to be, but I can equally believe that in a remote execution context this may be a docker bind mount, or even an NFS share or similar.

Reply:
Good point: thanks!